lib/Conversion/TTIRToTTNN/TTIRToTTNN.cpp

```diff
@@ -503,9 +506,19 @@ class ConstantOpConversionPattern
 
     if (valueAttr.isSplat()) {
       Value device = getOrInsertDevice(rewriter, op);
-      float fillValue = valueAttr.getElementType().isInteger()
-                            ? static_cast<float>(valueAttr.getSplatValue<int>())
-                            : valueAttr.getSplatValue<float>();
+      float fillValue = 0;
+      if (op.getResult().getType().getElementType().isBF16()) {
+        uint32_t raw_shifted_bfloat = valueAttr.getSplatValue<uint32_t>() << 16;
+        fillValue = *reinterpret_cast<float *>(&raw_shifted_bfloat);
```

Comment:
I don't understand how this works, can you add a comment to explain it please?

Reply:
Yeah so since bfloat16 isn't a type I can just have I need to just move around the bits representing the scalar value here I get the splat value as a uint32 so that I can perform a bit shift. The bfloat16 bits will be in the right-hand 16 bits of the uint32_t. A bfloat16 is the same as a float32 but with a much smaller mantissa. So, bit shifting this 16 bits to the right performs the cast to float32. I need to use a reinterpret cast on a pointer so that the bits remain the same.

Reply:
Looks like this PR will solve this issue for me actually so don't worry about my floating-point bit-shifting hack: https://github.com/tenstorrent/tt-mlir/pull/1164